Source/WebCore/style/values/grid/StyleGridTemplateAreas.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2025 Samuel Weinig <sam@webkit.org>
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include "CSSGridTemplateAreas.h"
+#include "StyleGridNamedAreaMap.h"
+#include "StyleGridNamedLinesMap.h"
+
+namespace WebCore {
+namespace Style {
+
+// <'grid-template-areas'> = none | <string>+
+// https://drafts.csswg.org/css-grid/#propdef-grid-template-areas
+struct GridTemplateAreas {
+    GridNamedAreaMap map;
+    GridNamedLinesMap implicitNamedGridColumnLines;
+    GridNamedLinesMap implicitNamedGridRowLines;
+
+    GridTemplateAreas(CSS::Keyword::None)
+        : map { }
+        , implicitNamedGridColumnLines { }
+        , implicitNamedGridRowLines { }
+    {
+    }
+
+    explicit GridTemplateAreas(GridNamedAreaMap&&);
+    explicit GridTemplateAreas(const GridNamedAreaMap&);
```

Comment:
Other structures like this have implicit constructors rather than `explicit`. Why a different choice here?